main_test.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestUsage(t *testing.T) {
+	w := bytes.NewBuffer(nil)
+	flag.CommandLine.SetOutput(w)
+	usage()
+	opts := `[-p <port>] [-a] [<dir>]`
+	desc := `Simple HTTP server, serving files from given directory.`
+	require.Contains(t, w.String(), opts)
+	require.Contains(t, w.String(), desc)
+}
+
+func TestDir(t *testing.T) {
+	require.Equal(t, ".", dir(nil))
+	require.Equal(t, ".", dir([]string{}))
+	require.Equal(t, "abc", dir([]string{"abc"}))
+	require.Equal(t, "abc", dir([]string{"abc", "efg"}))
+}
+
+func TestListenAddr(t *testing.T) {
+	require.Equal(t, ":0", listenAddr(0, true))
```

Comment:
the listenaddr function does not discriminate on port number or handle 0 as a special case. Does it make sense to have tests for both 0 and 8080? The logic for port 0 being any port is elsewhere.

Reply:
ok, removed.